docs/source/examples/e2e_flow.rst

```diff
@@ -0,0 +1,438 @@
+
+
+.. code-block:: bash
+
+    [generate.py:92] Exploratorium in San Francisco has made the cover of Time Magazine,
+                     and its awesome. And the bridge is pretty cool...
+
+    [generate.py:96] Time for inference: 11.61 sec total, 25.83 tokens/sec
+    [generate.py:99] Memory used: 15.72 GB
+
+
+Indeed, the bridge is pretty cool! Seems like our LLM knows what it's talking
+about.
+
+|
+
+Speeding up Generation using Quantization
+-----------------------------------------
+
+We saw that the generation recipe took around 11.6 seconds to generate 300 tokens.
+One technique commonly used to speed up inference is quantization. TorchTune provides
+an integration with the TorchAO quantization APIs. Let's first quantize the model using
+4-bit weights-only quantization and see if this improves generation.
+
+
+For this, we'll use the
+`quantization recipe <https://github.com/pytorch/torchtune/blob/main/recipes/quantize.py>`_.
+
+
+Let's first copy over the config to our local working director so we can make changes.
+
+.. code-block:: bash
+
+    tune cp quantize ./custom_quantization_config.yaml
+
+Let's modify ``custom_quantization_config.yaml`` to include the following changes.
+
+.. code-block:: yaml
+
+    checkpointer:
+        _component_: torchtune.utils.FullModelHFCheckpointer
+
+        # directory with the checkpoint files
+        # this should match the output_dir specified during
+        # finetuning
+        checkpoint_dir: <checkpoint_dir>
+
+        # checkpoint files for the fine-tuned model. This should
+        # match what's shown in the logs above
+        checkpoint_files: [
+            hf_model_0001_0.pt,
+            hf_model_0002_0.pt,
+        ]
+
+        output_dir: <checkpoint_dir>
+        model_type: LLAMA2
+
+
+Once the config is updated, let's kick off quantization! We'll use the default
+quantization method from the config.
+
+
+.. code-block:: bash
+
+    tune run quantize --config quantize
+
+Once quantization is complete, you'll see the following in the logs.
+
+.. code-block:: bash
+
+    [quantize.py:68] Time for quantization: 19.76 sec
+    [quantize.py:69] Memory used: 13.95 GB
+    [quantize.py:82] Model checkpoint of size 3.67 GB saved to <checkpoint_dir>/hf_model_0001_0-4w.pt
+
+
+.. note::
+    Unlike the fine-tuned checkpoints, this output a single checkpoint file. This is
+    because our quantization APIs currently don't support any conversion across formats.
+    As a result you won't be able to use these quantized models outside of TorchTune.
+    But you should be able to use these with the generation and evaluation recipes within
+    TorchTune. These results will help inform which quantization methods you should use
+    with your favorite inference engine.
+
+Now that we have the quantized model. Let's rerun generation.
+
+Let's modify ``custom_generation_config.yaml`` to include the following changes.
+
+.. code-block:: yaml
+
+    checkpointer:
```

Comment:
Just curious, I am seeing a lot of checkpointer across all these various tasks. Wondering if for some of the cases we want to just override things via CLI? E.g. in some cases (though not here) can't we just set `checkpointer.checkpoint_dir=<checkpoint_dir>` and call it a day? Or is that too much of a black box/defeats the purpose of showing off `tune cp`

Reply:
Nope this is a good point. I modifed this in the first one but then leaving as is when we have a ton of files to change